recipes/llvm/all/conanfile.py

```
@@ -0,0 +1,329 @@
+            'LLVM_BUILD_LLVM_DYLIB': self.options.shared,
+            'LLVM_DYLIB_COMPONENTS': self.options.components,
+            'LLVM_ENABLE_PIC': self.options.get_safe('fPIC', default=False),
+            'LLVM_ABI_BREAKING_CHECKS': 'WITH_ASSERTS',
+            'LLVM_ENABLE_WARNINGS': True,
+            'LLVM_ENABLE_PEDANTIC': True,
+            'LLVM_ENABLE_WERROR': False,
+            'LLVM_TEMPORARILY_ALLOW_OLD_TOOLCHAIN': True,
+            'LLVM_USE_RELATIVE_PATHS_IN_DEBUG_INFO': False,
+            'LLVM_BUILD_INSTRUMENTED_COVERAGE': False,
+            'LLVM_OPTIMIZED_TABLEGEN': True,
+            'LLVM_REVERSE_ITERATION': False,
+            'LLVM_ENABLE_BINDINGS': False,
+            'LLVM_CCACHE_BUILD': False,
+            'LLVM_INCLUDE_TOOLS': self.options.shared,
+            'LLVM_INCLUDE_EXAMPLES': False,
+            'LLVM_INCLUDE_TESTS': False,
+            'LLVM_INCLUDE_BENCHMARKS': False,
+            'LLVM_APPEND_VC_REV': False,
+            'LLVM_BUILD_DOCS': False,
+            'LLVM_ENABLE_IDE': False,
+            'LLVM_ENABLE_TERMINFO': False,
+            'LLVM_ENABLE_EH': self.options.exceptions,
+            'LLVM_ENABLE_RTTI': self.options.rtti,
+            'LLVM_ENABLE_THREADS': self.options.threads,
+            'LLVM_ENABLE_LTO': self.options.lto,
+            'LLVM_STATIC_LINK_CXX_STDLIB': self.options.static_stdlib,
+            'LLVM_ENABLE_UNWIND_TABLES': self.options.unwind_tables,
+            'LLVM_ENABLE_EXPENSIVE_CHECKS': self.options.expensive_checks,
+            'LLVM_ENABLE_ASSERTIONS': self.settings.build_type == 'Debug',
+            'LLVM_USE_NEWPM': False,
+            'LLVM_USE_OPROFILE': False,
+            'LLVM_USE_PERF': self.options.use_perf,
+            'LLVM_ENABLE_Z3_SOLVER': False,
+            'LLVM_ENABLE_LIBPFM': False,
+            'LLVM_ENABLE_LIBEDIT': False,
+            'LLVM_ENABLE_FFI': self.options.with_ffi,
+            'LLVM_ENABLE_ZLIB': self.options.get_safe('with_zlib', False),
```

We should set these variables to `FORCE_ON` instead of True so they add the `REQUIRED` keyword. Otherwise if CMake fails to find the dependencies they get silently ignored.

---

This is the code form version 16.0.03
```cmake
if(LLVM_ENABLE_ZLIB)
  if(LLVM_ENABLE_ZLIB STREQUAL FORCE_ON)
    find_package(ZLIB REQUIRED)
  elseif(NOT LLVM_USE_SANITIZER MATCHES "Memory.*")
    find_package(ZLIB)
  endif()
  if(ZLIB_FOUND)
  ...
```

---

Its also there in in llvm 13 and documentation showed that this is the case for other options as well. Can add it in my pr.